### DISCUSSION TITLE
Optionally expect heartbeats from the client

### DISCUSSION THREAD

**Discussion by majek** [sockjs/sockjs-node#290] [Ideas]
We should have an opt-in mechanism for enabling verifying heartbeats on the server.

With this option enabled the client MUST send heartbeat frames once in a while, so this involves changes in sockjs-client.

**Comment by glasser** [comment]
There was a lot of activity on this issue but it's still open. Does the feature work (and it should be closed)?

**Comment by brycekahle** [comment]
I would prefer this be a ping-pong type mechanism instead of just a timed interval of expected data from the client. The server pings every X seconds, and expects a pong within X seconds. If not, the connection is considered disconnected.

**Comment by brycekahle** [comment]
I just added support for this in websockets (see #169). Other protocols need to follow.
